src/main/java/eu/miaplatform/decorators/predecorators/PreDecoratorRequestProxy.java

```diff
@@ -0,0 +1,38 @@
+package eu.miaplatform.decorators.predecorators;
+
+import eu.miaplatform.decorators.DecoratorRequest;
+
+import java.util.Map;
+
+public class PreDecoratorRequestProxy {
+    public static class Builder {
+        private final DecoratorRequest request;
+
+        public Builder(DecoratorRequest originalRequest) {
+            this.request = originalRequest;
+        }
+
+        public Builder setBody(Object body) {
+            this.request.setBody(body);
+            return this;
+        }
+
+        public Builder setHeaders(Map<String, String> headers) {
+            this.request.setHeaders(headers);
+            return this;
+        }
+
+        public Builder setQuery(Map<String, String> query) {
+            this.request.setQuery(query);
+            return this;
+        }
+
+        public PreDecoratorRequest build() {
+            return PreDecoratorRequest.builder().request(request).build();
+        }
+    }
+
+    private PreDecoratorRequestProxy() {
```

Comment:
Why making the constructor private? couldn't this be a static or abstract class?

Reply:
I used a private constructor to prevent direct instantiation of the PreDecoratorRequestProxy class.
An outer class cannot be static and making it abstract wouldn't work because another class could inherit and implement it.